src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  const stateClone = Object.assign({}, state);
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, action.extraData);
+      history.push({ ...stateClone });
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
```

Comment:
I suggest, as was suggested to me by a mentor, that you create a function that can iterate over an array of keys, and delete them from your stateClone, so that you dont have to have 2 loops in both your `clear` and `removeProperties` action types.

Reply:
Could you help me understand how it's supposed to be done? To be honest I don't have an idea of what are you talking about.